feat: add campaign callout list element

marjisound opened the pull request:
Paired on this: @abeddow91 & @marjisound & @jonathonherbert 
## What does this change?
<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a Trello card over linking to it; the card may not always exist and reviewers may not have access to the board. -->
This PR creates a new prosemirror element `campaign-callout-list` through which would fetch the campaigns using the `fetchCampaignList` prop and render a dropdown showing the callouts. 

![image](https://user-images.githubusercontent.com/15894063/192546156-807f6788-8865-4641-97c5-139ac5cd06b8.png)

When the user chooses one of the callouts, the `callout` element shows up with the details about the callout. 
![image](https://user-images.githubusercontent.com/15894063/192546303-fee84c9a-f1b8-47ed-8765-31c5e3b29ba7.png)

Whilst pairing we also noticed that there were improper imports of `undefinedDropdownValue` from `membershipSpec`. We have removed this export in favour of always importing from `helpers/transform`. 

## How to test
<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->

1- The component was added to the demo project, so was tested by running `yarn start` and playing around with it.

2- This component was also tested with composer locally

TODO:
Needs to be tested in CODE env

## Accessibility
<!-- Usually only applicable to UI changes, check the boxes if you are satisfied that your changes pass these tests. 

This repository follows the Editorial Tools accessibility guidelines: https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md
-->


- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [ ] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [ ] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [ ] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [ ] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

jonathonherbert commented on demo/index.ts:
```diff
@@ -131,6 +135,69 @@ const mockThirdPartyTracking = (html: string) =>
         reach: { unsupportedPlatforms: [] },
       });
 
+const mockCampaignList = [
```
Let's put this in `sampleElements`, as it's effectively fixture data.

jonathonherbert reviewed (approved):
Looks great. A few non-block comments –

- Does it make sense to rename `CampaignCalloutListSpec.tsx` -> `Callout.tsx`?
- It'd be nice to consolidate the types related to callouts (`Fields`, `Rules`, `Campaign`) with the existing callout components, either in this PR or a follow-up.

marjisound commented:
Thanks @jonathonherbert for looking at the PR. We just addressed 2 of your comments regarding the name changing. 
We also added the `html` to the fields so that can be used for composer preview. 
Could you please have another look at the changes we did. Thank you 

> It'd be nice to consolidate the types related to callouts (Fields, Rules, Campaign) with the existing callout components, either in this PR or a follow-up.
We'll do this in a separate PR if that's OK with you.

jonathonherbert commented:
Yes, that's fine. For other reviewers, it might be worth noting – am I right? – that this is effectively a draft version of this element – the fields especially may change!

marjisound commented:
> 



> Yes, that's fine. For other reviewers, it might be worth noting – am I right? – that this is effectively a draft version of this element – the fields especially may change!

Yes that's true. This version is not going to be used by editorials. We are going to have another PR that will finalise the new Callout component with updated fields 👍

jonathonherbert reviewed (commented):
Changes look good!

One thing to mention – we should not need to store any html to preview the element in Composer. We can write some code in Composer to come up with a suitable preview, based on the campaign id.

In the medium term, it'd be nice to be able to express both the editable and preview states from within `prosemirror-elements` – this is a good reminder that at the moment, for people creating new elements, we cannot define that behaviour in this repo.

Happy to talk through the above if anything's not clear.
